src/key.js

```diff
@@ -624,9 +624,83 @@ async function mergeSignatures(source, dest, attr, checkFn) {
   }
 }
 
-// TODO
-Key.prototype.revoke = function() {
+/**
+ * Revokes the key
+ * @param  {module:key~Key} privateKey decrypted private key for revocation
+ * @param  {Object} reasonForRevocation optional, object indicating the reason for revocation
+ * @param  {module:enums.reasonForRevocation} reasonForRevocation.flag optional, flag indicating the reason for revocation
+ * @param  {String} reasonForRevocation.string optional, string explaining the reason for revocation
+ * @param  {Date} date optional, override the creationtime of the revocation signature
+ * @return {module:key~Key} new key with revocation signature
+ */
+Key.prototype.revoke = async function(privateKey, {
+  flag: reasonForRevocationFlag=enums.reasonForRevocation.no_reason,
+  string: reasonForRevocationString=''
+} = {}, date=new Date()) {
+  if (privateKey.isPublic()) {
+    throw new Error('Need private key for revoking');
+  }
+  if (privateKey.primaryKey.getFingerprint() !== this.primaryKey.getFingerprint()) {
+    throw new Error('Private key does not match public key');
+  }
+  const dataToSign = { key: this.primaryKey };
+  const key = new Key(this.toPacketlist());
+  key.revocationSignatures.push(await createSignaturePacket(dataToSign, privateKey, privateKey.primaryKey, {
+    signatureType: enums.signature.key_revocation,
+    reasonForRevocationFlag: enums.write(enums.reasonForRevocation, reasonForRevocationFlag),
+    reasonForRevocationString
+  }, date));
+  return key;
+};
 
+/**
+ * Get revocation certificate from a revoked key.
+ *   (To get a revocation certificate for an unrevoked key, call revoke() first.)
+ * @return {String} armored revocation certificate
+ */
+Key.prototype.getRevocationCertificate = function() {
+  if (this.revocationSignatures.length) {
+    const commentstring = config.commentstring;
+    config.commentstring = 'This is a revocation certificate';
```

Comment:
If I make a revocation cert in GPG I get two comment lines:
```
Comment: GPGTools - https://gpgtools.org
Comment: This is a revocation certificate
```
So instead of reassigning the comment and doing this try/finally stuff I would just pass a flag indicating the revocation into the armor.encode function.